packages/cli/src/commands/database/seed/tables.ts

```diff
@@ -0,0 +1,127 @@
+import { readdir, readFile } from 'fs/promises';
+import path from 'path';
+
+import { generateStandardId } from '@logto/core-kit';
+import {
+  managementResource,
+  defaultSignInExperience,
+  createDefaultAdminConsoleConfig,
+  createDemoAppApplication,
+  defaultRole,
+  managementResourceScope,
+  defaultRoleScopeRelation,
+  defaultTenantId,
+} from '@logto/schemas';
+import { Hooks, Tenants } from '@logto/schemas/models';
+import type { DatabaseTransactionConnection } from 'slonik';
+import { sql } from 'slonik';
+import { raw } from 'slonik-sql-tag-raw';
+
+import { insertInto } from '../../../database.js';
+import { getDatabaseName } from '../../../queries/database.js';
+import { updateDatabaseTimestamp } from '../../../queries/system.js';
+import { getPathInModule } from '../../../utilities.js';
+import { createTenant } from './tenant.js';
+
+const getExplicitOrder = (query: string) => {
+  const matched = /\/\*\s*init_order\s*=\s*([\d.]+)\s*\*\//.exec(query)?.[1];
+
+  return matched ? Number(matched) : undefined;
+};
+
+const compareQuery = ([t1, q1]: [string, string], [t2, q2]: [string, string]) => {
+  const o1 = getExplicitOrder(q1);
+  const o2 = getExplicitOrder(q2);
+
+  if (o1 === undefined && o2 === undefined) {
+    return t1.localeCompare(t2);
+  }
+
+  if (o1 === undefined) {
+    return 1;
+  }
+
+  if (o2 === undefined) {
+    return -1;
+  }
+
+  return o1 - o2;
+};
+
+type Lifecycle = 'before_all' | 'after_all' | 'after_each';
+
+const lifecycleNames: readonly string[] = Object.freeze([
+  'before_all',
+  'after_all',
+  'after_each',
+] satisfies Lifecycle[]);
+
+export const createTables = async (connection: DatabaseTransactionConnection) => {
+  const tableDirectory = getPathInModule('@logto/schemas', 'tables');
+  const directoryFiles = await readdir(tableDirectory);
+  const tableFiles = directoryFiles.filter((file) => file.endsWith('.sql'));
+  const queries = await Promise.all(
+    tableFiles.map<Promise<[string, string]>>(async (file) => [
+      file,
+      await readFile(path.join(tableDirectory, file), 'utf8'),
+    ])
+  );
+
+  const runLifecycleQuery = async (
+    lifecycle: Lifecycle,
+    parameters: { name?: string; database?: string } = {}
```

Comment:
According to lifecycle SQLs, It seems that the database field is required.

Reply:
It might not be the most suitable approach to utilize a single function for running three life cycle scripts, as they have distinct parameter requirements.

Reply:
I think this function can be the fundamental one and the specific lifecycle query runner is just a wrapper of this one

Reply:
i'm good to make database required